src/Components/Modules/Weapon.cpp

```diff
@@ -520,6 +521,32 @@ namespace Components
 		GSC::Script::AddMethod("FreezeControlsAllowLook", PlayerCmd_FreezeControlsAllowLook);
 	}
 
+
+	typedef void (*PM_Weapon_t)(Game::pmove_s* pm, Game::pml_t* pml);
+	PM_Weapon_t PM_Weapon = reinterpret_cast<PM_Weapon_t>(0x44C380);
+	;
+
+	void PM_Weapon_stub(Game::pmove_s* pm, Game::pml_t* pml)
+	{
+		if (Weapon::BGMW3Swapping->current.enabled)
+		{
+			int state = pm->ps->weapState[0x0].weaponState;
+			if (pm->cmd.weapon == pm->ps->weapCommon.weapon &&
+				(state == Game::WEAPON_DROPPING ||
+					state == Game::WEAPON_DROPPING_QUICK ||
+					state == Game::WEAPON_DROPPING_ALT))
+			{
+				for (int index = 0; index < 2; index++)
+				{
+					pm->ps->weapState[index].weaponState = 1;
+					pm->ps->weapState[index].weaponTime = Game::BG_GetWeaponDef(pm->ps->weapCommon.weapon)->quickRaiseTime;
+					pm->ps->weapState[index].weapAnim = 20;
+				}
```

Comment:
```suggestion
				for (int i = 0; i < 2; i++)
				{
					pm->ps->weapState[i].weaponState = 1;
					pm->ps->weapState[i].weaponTime = Game::BG_GetWeaponDef(pm->ps->weapCommon.weapon)->quickRaiseTime;
					pm->ps->weapState[i].weapAnim = 20;
				}
```

Reply:
1 and 20 are mysterious constants, someone reading this will think they're secret cheat codes. At minimum, define constants like WEAPON_ACTIVE or ANIM_QUICK_RAISE

Reply:
FWIW it's not clear what 2 represent here, but assuming it's weapon state, we can do

```suggestion
			std::for_each (std::begin (pm->ps->weapState), std::end (pm->ps->weapState),
					[&] (auto &ws)
				{
					ws.weaponState = 1;
					ws.weaponTime = Game::BG_GetWeaponDef (pm->ps->weapCommon.weapon)->quickRaiseTime;
					ws.weapAnim = 20;
				});
```